lib/state.js

```diff
@@ -569,6 +569,82 @@ class State {
 
     return result.rowCount;
   }
+   
+   /**
+   * Insert an AMI's first usage.
+   */
+  async insertAmiUsage({region, id}, client) {
+    assert(typeof region === 'string');
+    assert(typeof id === 'string');
+
+    let text = [
+      "INSERT INTO amiusage (region, id, lastused)",
+      "VALUES ($1, $2, CURRENT_TIMESTAMP)"
+    ].join(' ');
+    let values = [region, id];
+    let name = 'insert-amiusage';
+
+    let result;
+    if (client) {
+      result = await client.query({text, values, name});
+    } else {
+      result = await this._pgpool.query({text, values, name});
+    }
```

Comment:
maybe better:
```javascript
insertAmiUsage(region, id, client) {
  assert(typeof region === 'string');
  assert(typeof id === 'string');

  const query = client ? client.query : this._pgpool.query;

  return query({
    text: `INSERT INTO amiusage (region, id, lastused) VALUES (${region}, ${id}, CURRENT_TIMESTAMP)`,
    name: 'insert-amiusage'
  }).then(result => assert.equal(
    result.rowCount,
    1,
    'inserting AMI usage had incorrect rowCount'
  ));
}
```

This is more compact, uses less temporary variables and it is more clear that the intention of the `if-then-else` clause just boils down to choosing which `query` method to use. Although I didn't use here, the `[].join(' ')` trick is welcome if the text lines turns out to be too long.

Reply:
I think it's probably better for consistency here if we stick to using async and await.